File: Algorithm-Swift/Algorithm-Swift/Inflearn/14_뒤집은소수.swift
//
//  14_뒤집은소수.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/29.
//

import Foundation

func isPrime2(_ num: Int) -> Bool {
    if num == 1 { return false }

    var cnt = 0

    for i in 1..<num/2+1 {
        if num % i == 0 {
            cnt += 1
        }
    }

    return cnt <= 1 ? true : false
}

func reverse(_ num: Int) -> Int {
    var num = num
    var temp = ""

    while num != 0 {
        temp += "\(num % 10)"
        num /= 10
    }

    return Int(temp)!
}

func solution14() {
    let _ = Int(readLine()!)!
    let nums = readLine()!.split(separator: " ").map { Int($0)! }
    var reverses = ""

    nums.forEach {
        let reverseNum = reverse($0)
        if isPrime2(reverseNum) {
            reverses += "\(reverseNum) "
        }
    }

    print(reverses)
}

File: Algorithm-CPP/14_뒤집은소수.cpp
#include <iostream>
#include <string>
using namespace std;

int isPrime(int x) {
    if(x <= 1) { return 0; }

    int cnt = 0;
    for(int i=1; i<(x/2 + 1); i++) {
        if(x % i == 0) {
            cnt++;
        }
    }

    if(cnt <= 1) return 1;
    else return 0;
}

int reverse(int x) {
    string n;

    while(x != 0) {
        int num = x % 10;
        x /= 10;
        n += to_string(num);
    }

    return stoi(n);
}

int main() {
    int n;
    string result;

    cin >> n;

    for(int i=0; i<n; i++) {
        int num;
        cin >> num;

        int reverseNum = reverse(num);
        if (isPrime(reverseNum)) {
            result += to_string(reverseNum) + ' ';
        }
    }

    cout << result << endl;
}
